test/test_objectives.py

```diff
@@ -20,20 +22,54 @@
 )
 
 
+""" Tests for distance objective """
+
+
 @pytest.mark.parametrize(
-    "method",
+    "method,max_motion_cost",
     [
-        objectives.DistanceMethod.EUCLIDEAN,
-        objectives.DistanceMethod.LATLON,
-        objectives.DistanceMethod.OMPL_PATH_LENGTH,
+        (objectives.DistanceMethod.EUCLIDEAN, 2.5),
+        (objectives.DistanceMethod.LATLON, 2700),
+        (objectives.DistanceMethod.OMPL_PATH_LENGTH, 4.0),
     ],
 )
-def test_distance_objective(method: objectives.DistanceMethod):
+def test_distance_objective(method: objectives.DistanceMethod, max_motion_cost: float):
     distance_objective = objectives.DistanceObjective(
         PATH._simple_setup.getSpaceInformation(),
         method,
     )
-    assert distance_objective is not None
+
+    # test sample_states()
+    num_samples = 3
+    sampled_states = distance_objective.sample_states(num_samples)
+    assert len(sampled_states) == num_samples
+    # for state in sampled_states:
+    #     assert ompl_path.is_state_valid(state)
```

Comment:
We discussed this a while ago, but I think sample_states should return valid states. Then you can uncomment these 2 lines from all the new tests

Reply:
Sure, but it will fail cause the validity check is not implemented in the sampler. I think Sean has to first define the target boats and islands for us to implement this. Let me know if I am wrong. It can easily be added I just have to add that check before appending to the list within sampler function.

Reply:
You can use the function in this commented out code, `is_state_valid()`

Reply:
Yea no need to do this as we discussed, the cost doesn't take into account the state validity anyways